activerecord/lib/active_record/fixture_set/file.rb

```diff
@@ -66,7 +66,7 @@ def render(content)
         # Validate our unmarshalled data.
         def validate(data)
           unless Hash === data || YAML::Omap === data
-            raise Fixture::FormatError, "fixture is not a hash"
+            raise Fixture::FormatError, "fixture is not a hash: #{@file}"
           end
 
           raise Fixture::FormatError unless data.all? { |name, row| Hash === row }
```

Comment:
Is it worth adding some more info to this error too while we're here? Looks like it could name both the filename and the `name` key.